src/main/java/org/springframework/data/hazelcast/repository/query/HazelcastSortAccessor.java

```diff
@@ -72,7 +72,11 @@
             HazelcastPropertyComparator hazelcastPropertyComparator = new HazelcastPropertyComparator(order.getProperty(),
                     order.isAscending());
 
-            compoundComparator.addComparator(hazelcastPropertyComparator);
+            if (compoundComparator == null) {
+                compoundComparator = hazelcastPropertyComparator;
+            } else {
+                compoundComparator.thenComparing(hazelcastPropertyComparator);
```

Comment:
Then would not have worked since `thenComparing` returns a new `Comparator` instance and doesn't mutate the original